### PR TITLE
Yaakov v2

### PR DESCRIPTION
**Pull request by ybressler**
**Preprocessing is complete.**
- There are several heuristic approaches taken which I'd love some input on.
- Also, data in the newest s3 bucket isn't up to date with the displayed survey for reporting a diagnosis. _Need to see that data before determining if existing code is satisfactory._ 

**Oversample method is complete:**
- SMOTE method.
- Modified code to make more robust.
- Works with very small sample sizes. Min N (of Y, either True of False) = 4!

### CODE REVIEW

**Comment by rashnil-git** [comment]
Thanks @ybressler : I will review sometime this week. The data on s3 is the latest. There is no new data reported through the UI. We have 13 records as of now. Whatever is being sent from the survey app goes to DB. We are collecting data from UI based on what the survey app is sending to us.

**Comment by rashnil-git** [comment]
@ybressler : It's good. I recommend we create a new folder `utils` under v2 and keep the files that are supplement there. For example, date time, AWS creds etc so that they can me used in different modules/classes. Also, instead of giving the name main.py - we can give some meaningful name based on functionality.

File: v2/standard_functions/datetime.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "|" + #Or
+        "[0-9]{2}-[0-9]{2}-[0-9]{4}"
+    )
+    dt = pattern.search(string).group()
+
+    if re.search("[0-9]{4}-[0-9]{2}-[0-9]{2}",dt):
+        dt = datetime.datetime.strptime(dt,"%Y-%m-%d")
+    else:
+        dt = datetime.datetime.strptime(dt,"%m-%d-%Y")
+    return dt
```
Comment (rashnil-git) [inline v2/standard_functions/datetime.py]: lets rename this to utils instead of standardized functions. ?

File: v2/preprocess_data.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
+# PURPOSE: This program preprocesses existing data for machine learning training.
```
Comment (rashnil-git) [inline v2/preprocess_data.py]: lets rename it to `user_reports_preprocessor.py`? We might have other jobs in future so was thinking this name might be very generic?

**Review by rashnil-git — APPROVED** [review]
Overall looks good, except some minor changes to folder structure and naming.

File: v2/standard_functions/datetime.py (hunk 1 above)
Comment (ybressler) [reply v2/standard_functions/datetime.py]: Done.

File: v2/preprocess_data.py (hunk 2 above)
Comment (ybressler) [reply v2/preprocess_data.py]: Sure.

**Pull request merged**